ocf_data_sampler/load/satellite.py

```diff
@@ -42,35 +60,160 @@ def get_single_sat_data(zarr_path: str) -> xr.Dataset:
+            session = repo.readonly_session(version=commit_sha)
+            logger.info("Successfully opened session using 'version' parameter")
+        except (TypeError, AttributeError, Exception):
+            pass
+        
+        # Method 2: If version doesn't work, try accessing via the main branch 
+        # (since the commit should be the latest on main)
+        if not session:
+            try:
+                session = repo.readonly_session(branch="main")
+                logger.info("Opened main branch (commit should be latest on main)")
+            except Exception as e:
+                raise ValueError(f"Could not open session for commit {commit_sha}: {e}")
+    else:
+        logger.info("Opening 'main' branch of Ice Chunk repository.")
+        session = repo.readonly_session(branch="main")
+
+    # Open the dataset from the Ice Chunk session store
+    ds = xr.open_zarr(session.store, consolidated=True, chunks="auto")
+
+    # Select specific channels if requested
+    if channels:
+        available_channels = [ch for ch in channels if ch in ds.data_vars]
+        if not available_channels:
+            raise ValueError(f"None of the requested channels {channels} found in the dataset.")
+        logger.info(f"Selecting channels: {available_channels}")
+        ds = ds[available_channels]
+
+    # Combine separate channel variables into a single DataArray
+    data_arrays = [ds[var] for var in sorted(ds.data_vars)]
+    da = xr.concat(data_arrays, dim="channel")
+    da = da.assign_coords(channel=sorted(ds.data_vars))
+
+    # Standardize coordinate names and structure
+    da = da.rename({"time": "time_utc"})
+    check_time_unique_increasing(da.time_utc)
+    da = make_spatial_coords_increasing(da, x_coord="x_geostationary", y_coord="y_geostationary")
+    da = da.transpose("time_utc", "channel", "x_geostationary", "y_geostationary")
+
+    return da
+
+def _open_sat_data_zarr(
+    zarr_path: str | list[str], channels: Optional[List[str]] = None
+) -> xr.DataArray:
+    """Lazily opens a standard zarr store, potentially with multiple files."""
     if isinstance(zarr_path, list | tuple):
         ds = xr.combine_nested(
-            [get_single_sat_data(path) for path in zarr_path],
+            [_get_single_sat_data_zarr(path) for path in zarr_path],
             concat_dim="time",
             combine_attrs="override",
             join="override",
         )
     else:
-        ds = get_single_sat_data(zarr_path)
+        ds = _get_single_sat_data_zarr(zarr_path)
+
+    if channels:
+        ds = ds.sel(variable=channels)
 
     ds = ds.rename(
         {
             "variable": "channel",
             "time": "time_utc",
         },
     )
-
     check_time_unique_increasing(ds.time_utc)
     ds = make_spatial_coords_increasing(ds, x_coord="x_geostationary", y_coord="y_geostationary")
     ds = ds.transpose("time_utc", "channel", "x_geostationary", "y_geostationary")
```

Comment:
This should be renamed back to just open_sat_data as the logic splitting between icechunks and regular zarrs gets moved by [1]